client/client.go

```diff
@@ -141,6 +141,9 @@ type Config struct {
 
 	// UserAgent is the User-Agent header sent to the Pebble daemon.
 	UserAgent string
+
+	// PebbleDir is the base directory for storing the Pebble daemon's state.
+	PebbleDir string
```

Comment:
I really don't think this (or the new exported methods) belong in the Client. It seems like with this change we're trying to use the Client instance as a way to pass through extra things that the client itself doesn't need. (I'm aware of the link between `PEBBLE` and `PEBBLE_SOCKET`, but I'm talking about code structure here.) So I think we should revert these client changes and find a better way to give the relevant commands access to these settings without passing them through here.

It seems reasonable to add `PebbleDir string` to `RunOptions` (it is configuration for `Run()`), and then add a `RunOptions *RunOptions` field to `CmdOptions`, which is (more or less) accessible to the commands. Then `cmdRun` would have a `pebbleDir string` field, and `cmdNotices` and `cmdOkay` would have a `pebbleSocket string` field. It's a tiny bit more wiring, but definitely feels better than threading them through client.

Open to alternative suggestions too!

The notices code also uses `$XDG_CONFIG_HOME` and so we should probably wire up a `CLIStatePath` option, but I'm very happy to leave that till if/when we need it.

Reply:
Passing `RunOptions` to `CmdOptions` and taking the values from there is better, yes.

I tried to do this now, unfortunately it seems like it requires passing `RunOptions` to `cli.Parse()`, and that results in a big diff due to tests using `cli.Parse()` directly quite often (see other comments).

Also, for some reason, the `enter` command (or its tests) seem to fail now (also see other comment there), didn't  immediately spot what's wrong there, maybe you see something there that I don't?